editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,450 @@
+	}
+}
+
+void GroupSettingsEditor::update_groups() {
+	if (updating_group) {
+		return;
+	}
+
+	updating_group = true;
+
+	groups_cache = ProjectSettings::get_singleton()->get_global_groups_list();
+
+	tree->clear();
+
+	TreeItem *root = tree->create_item();
+
+	List<StringName> keys;
+	for (const KeyValue<StringName, String> &E : groups_cache) {
+		keys.push_back(E.key);
+	}
+	keys.sort_custom<NoCaseComparator>();
+
+	for (const StringName &E : keys) {
+		TreeItem *item = tree->create_item(root);
+		item->set_meta("__name", E);
+		item->set_meta("__description", groups_cache[E]);
+
+		item->set_text(0, E);
+		item->set_editable(0, true);
+
+		item->set_text(1, groups_cache[E]);
+		item->set_editable(1, true);
+		item->add_button(2, get_theme_icon(SNAME("Remove"), SNAME("EditorIcons")));
+		item->set_selectable(2, false);
+	}
+
+	updating_group = false;
+}
+
+void GroupSettingsEditor::_confirm_delete() {
+	TreeItem *ti = tree->get_selected();
+	if (!ti) {
+		return;
+	}
+
+	String name = ti->get_text(0);
+	String description = groups_cache[name];
+	String property_name = GLOBAL_GROUP_PREFIX + name;
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Remove Group"));
+
+	undo_redo->add_do_property(ProjectSettings::get_singleton(), property_name, Variant());
+	undo_redo->add_undo_property(ProjectSettings::get_singleton(), property_name, description);
+
+	if (remove_check_box->is_pressed()) {
+		undo_redo->add_do_method(this, "remove_references", name);
+	}
+
+	undo_redo->add_do_method(this, "call_deferred", "update_groups");
+	undo_redo->add_undo_method(this, "call_deferred", "update_groups");
+
+	undo_redo->add_do_method(this, "emit_signal", group_changed);
+	undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+	undo_redo->commit_action();
+}
+
+void GroupSettingsEditor::show_message(const String &p_message) {
+	message->set_text(p_message);
+	message->popup_centered();
+}
+
+void GroupSettingsEditor::_add_button_pressed() {
+	_add_group(group_name->get_text(), group_description->get_text());
+}
+
+void GroupSettingsEditor::_changed() {
+	emit_signal(group_changed);
+	call_deferred("update_groups");
+}
+
+void GroupSettingsEditor::_show_remove_dialog() {
+	if (!remove_dialog) {
+		remove_dialog = memnew(ConfirmationDialog);
+		remove_dialog->connect("confirmed", callable_mp(this, &GroupSettingsEditor::_confirm_delete));
+
+		VBoxContainer *vbox = memnew(VBoxContainer);
+		remove_label = memnew(Label);
+		vbox->add_child(remove_label);
+
+		remove_check_box = memnew(CheckBox);
+		remove_check_box->set_text(TTR("Delete references from all scenes"));
+		vbox->add_child(remove_check_box);
+
+		remove_dialog->add_child(vbox);
+
+		add_child(remove_dialog);
+	}
+
+	TreeItem *ti = tree->get_selected();
+	if (!ti) {
+		return;
+	}
+
+	remove_check_box->set_pressed(false);
+	remove_label->set_text(vformat(TTR("Delete group \"%s\" and its description?"), ti->get_text(0)));
```

Comment:
```suggestion
	remove_label->set_text(vformat(TTR("Delete group \"%s\"?"), ti->get_text(0)));
```
No need to ask about description; it's part of the group (somewhat).